src/shell.rs

```diff
@@ -159,12 +159,39 @@ fn fixup_docker_container(container: &Container, u: u32, g: u32) -> LalResult<Co
     Ok(modified_container)
 }
 
-/// Runs an arbitrary command in the configured docker environment
+
+/// Runs an arbitrary command in the configured environment
+/// delegating to docker if needed.
+///
+/// This is the most general function, used by both `lal build` and `lal shell`.
+pub fn run(
+    cfg: &Config,
+    environment: &Environment,
+    command: Vec<String>,
+    flags: &DockerRunFlags,
+    modes: &ShellModes,
+) -> LalResult<()> {
+    match environment {
+        Environment::Container(container) => docker_run(&cfg, &container, command, &flags, &modes),
+
+        Environment::None => {
+            let mut command = command.clone();
+            let exe = command.remove(0);
+            let s = Command::new(exe).args(command).status()?;
+            if !s.success() {
+                return Err(CliError::SubprocessFailure(s.code().unwrap_or(1001)));
+            }
```

Comment:
tempted to have a `native_run` or something here, looks reused

Reply:
> `native_run`

I like that. I was going with `uncontained_run`, because I couldn't think of a word that didn't express itself as the opposite of something else.

In the future I can see us using other "runners". (https://firecracker-microvm.github.io/ hint hint)